src/app/Services/Frequency/Delete.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace LaravelEnso\Calendar\app\Services\Frequency;
+
+use Illuminate\Support\Facades\DB;
+use LaravelEnso\Calendar\app\Models\Event;
+use LaravelEnso\Calendar\app\Enums\UpdateType;
+
+class Delete extends Frequency
+{
+    public function handle($updateType)
+    {
+        if ($this->parent()->events->isEmpty()) {
+            return;
+        }
+
+        if ($updateType === UpdateType::All) {
+            Event::whereParentId($this->parent()->id)
+                ->orWhere('id', $this->parent()->id)
+                ->delete();
+
+            return;
+        }
+
+        if ($this->isParent() && $updateType === UpdateType::Single) {
+            $this->changeParent();
+        }
+    }
+
+    protected function changeParent()
+    {
+        $nextEventId = $this->parent()->events->first()->id;
```

Comment:
we should order by id or date to ensure the result is as we would expect it.

also, the nextEvent would be the next event after the deleted event, otherwise there could be cases where the retrieved event is something before the deleted one